packages/utils/src/activity.js

```diff
@@ -32,19 +32,32 @@ export function getAncestors(activities, activity) {
   return [...ancestors, parent];
 }
 
-export function getOutlineChildren(activities, schema, parentId) {
+export function getOutlineChildren(activities, parentId, schema) {
   const children = getChildren(activities, parentId);
   if (!parentId || !children.length) return children;
-  const types = schema.getLevel(find(activities, { id: parentId }).type).subLevels;
+  const parentType = find(activities, { id: parentId }).type;
+  const types = schema.getLevel(parentType).subLevels;
   return filter(children, it => types.includes(it.type));
 }
 
-export function toTreeFormat(activities, schema, parentId = null, level = 1) {
-  return getOutlineChildren(activities, schema, parentId).map(activity => ({
+export function getOutlineTree(activities, schema, options = {}) {
+  return toTreeFormat(activities, schema, { filterFn: getOutlineChildren, ...options });
```

Comment:
@underscope is this a good solution? 
I'm a little bit skeptical about passing down the filter function to the recursive function because the consumer will have obligation to pass the correct filter function which doesn't make infinite recursive calls. What do you think?

Reply:
We can have a depth parameter and throw an error if it's exceeded? 😉